auth/api/iam/openid4vp.go

```diff
@@ -97,6 +99,19 @@ func (r Wrapper) handleAuthorizeRequestFromHolder(ctx context.Context, verifier
 	if params.get(oauth.NonceParam) == "" {
 		return nil, withCallbackURI(oauthError(oauth.InvalidRequest, "missing nonce parameter"), redirectURL)
 	}
+	// we require PKCE (RFC7636) for authorization code flows
+	// check code_challenge and code_challenge_method
+	if params.get(oauth.CodeChallengeParam) == "" || params.get(oauth.CodeChallengeMethodParam) == "" {
```

Comment:
fail if `oauth.CodeChallengeMethodParam` is `plain`

[section 4.4.1](https://datatracker.ietf.org/doc/html/rfc7636#section-4.4.1)
> If the server supporting PKCE does not support the requested transformation, the authorization endpoint MUST return the authorization error response with "error" value set to "invalid_request".  The "error_description" or the response of "error_uri" SHOULD explain the nature of error, e.g., transform algorithm not supported.